src/test/java/com/mkdk72ki/asgmt10/integrationtest/UserRestApiIntegrationTest.java

```diff
@@ -0,0 +1,352 @@
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.Customization;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import org.skyscreamer.jsonassert.comparator.CustomComparator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.nio.charset.StandardCharsets;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@DBRider
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class UserRestApiIntegrationTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    // GET
+
+    @Test
+    @DataSet(value = "datasets/users.yml")
+    @Transactional
+    void ユーザーが全件取得できること() throws Exception {
+        String response = mockMvc.perform(MockMvcRequestBuilders.get("/users"))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+
+        JSONAssert.assertEquals("""
+                    [
+                    {
+                        "id": 1,
+                        "name": "山田太郎",
+                        "ruby": "yamada taro",
+                        "birthday": "1990-03-04",
+                        "email": "yamada@mkdk.com"
+                    },
+                    {
+                        "id": 2,
+                        "name": "山田花子",
+                        "ruby": "yamada hanako",
+                        "birthday": "2000-11-23",
+                        "email": "hanako@mkdk.com"
+                    },
+                    {
+                        "id": 3,
+                        "name": "小山田祐介",
+                        "ruby": "oyamada yusuke",
+                        "birthday": "2005-07-16",
+                        "email": "oyamada@mkdk.com"
+                    },
+                    {
+                        "id": 4,
+                        "name": "山田次郎",
+                        "ruby": "YAMADA JIRO",
+                        "birthday": "1995-12-30",
+                        "email": "jiro@mkdk.com"
+                    }
+                ]
+                                """, response, JSONCompareMode.STRICT);
+
+    }
+
+    @Test
+    @DataSet(value = "datasets/users.yml")
+    @Transactional
+    void ルビで指定したユーザーが取得できること() throws Exception {
+        String response = mockMvc.perform(MockMvcRequestBuilders.get("/users?ruby=yamada"))

```

Comment:
StartingWithをクエリパラメータにするほうがいいかもですね。
```suggestion
        String response = mockMvc.perform(MockMvcRequestBuilders.get("/users?rubyStartingWith=yamada"))

```
ただ、対応するにしてもマージ後時間があればでよいです。
今のままでも大丈夫 🆗

Reply:
かしこまりました。  
ひとまず今回はこのまま進めさせていただきます！